lib/rack/test/uploaded_file.rb

```diff
@@ -18,33 +18,28 @@ class UploadedFile
       # The content type of the "uploaded" file
       attr_accessor :content_type
 
-      def initialize(path, content_type = 'text/plain', binary = false)
-        raise "#{path} file does not exist" unless ::File.exist?(path)
-
+      def initialize(content, content_type = 'text/plain', binary = false, original_filename: nil)
+        if content.respond_to?(:read)
+          initialize_from_io(content, original_filename)
+        else
+          initialize_from_file_path(content)
+        end
         @content_type = content_type
-        @original_filename = ::File.basename(path)
-
-        @tempfile = Tempfile.new([@original_filename, ::File.extname(path)])
-        @tempfile.set_encoding(Encoding::BINARY) if @tempfile.respond_to?(:set_encoding)
         @tempfile.binmode if binary
-
-        ObjectSpace.define_finalizer(self, self.class.finalize(@tempfile))
-
-        FileUtils.copy_file(path, @tempfile.path)
       end
 
       def path
-        @tempfile.path
+        tempfile.path
       end
 
       alias local_path path
 
       def method_missing(method_name, *args, &block) #:nodoc:
-        @tempfile.__send__(method_name, *args, &block)
+        tempfile.public_send(method_name, *args, &block)
       end
 
-      def respond_to?(method_name, include_private = false) #:nodoc:
-        @tempfile.respond_to?(method_name, include_private) || super
+      def respond_to_missing?(method_name, include_private = false) #:nodoc:
+        tempfile.respond_to?(method_name, include_private) || super
```

Comment:
https://ruby-doc.org/core-2.4.1/Object.html#method-i-respond_to_missing-3F says:

> DO NOT USE THIS DIRECTLY.

Is there a way to implement this in another way?